src/main/java/net/minestom/server/extras/lan/OpenToLANConfig.java

```diff
@@ -0,0 +1,64 @@
+package net.minestom.server.extras.lan;
+
+import java.util.Objects;
+
+import net.minestom.server.event.server.ServerListPingEvent;
+import net.minestom.server.utils.time.TimeUnit;
+import net.minestom.server.utils.time.UpdateOption;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Configuration for opening the server to LAN.
+ * @see OpenToLAN#open(OpenToLANConfig)
+ */
+public class OpenToLANConfig {
+    int port;
+    UpdateOption delayBetweenPings, delayBetweenEvent;
+
+    /**
+     * Creates a new config with the port set to random and the delay between pings set
+     * to 1.5 seconds and the delay between event calls set to 30 seconds.
+     */
+    public OpenToLANConfig() {
+        this.port = 0;
+        this.delayBetweenPings = new UpdateOption(1500, TimeUnit.MILLISECOND);
+        this.delayBetweenEvent = new UpdateOption(30, TimeUnit.SECOND);
+    }
+
+    /**
+     * Sets the port used to send pings from. Use {@code 0} to pick a random free port.
+     *
+     * @param port the port
+     * @return {@code this}, for chaining
+     */
+    @Contract("_ -> this")
+    public @NotNull OpenToLANConfig setPort(int port) {
+        this.port = port;
+        return this;
+    }
+
+    /**
+     * Sets the delay between outgoing pings.
+     *
+     * @param delay the delay
+     * @return {@code this}, for chaining
+     */
+    @Contract("_ -> this")
+    public @NotNull OpenToLANConfig setDelayBetweenPings(@NotNull UpdateOption delay) {
```

Comment:
`#setDelay` would probably be enough

Reply:
Well we need to make it clear the difference between the delay between pings and the delay between event calls.

Reply:
Perhaps `#setPingDelay` would be better then?

Reply:
Also, if this is a sort of builder, then I think it would be a bit nicer if you omitted the `set` prefixes from these, as we aren't really changing state in the same way we would with a regular object, since we're using this object to build a new object.